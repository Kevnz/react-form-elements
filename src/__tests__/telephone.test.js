import React from 'react'
import { render, cleanup } from 'react-testing-library'

import { Telephone } from '../'

describe('Telephone component', () => {
  afterEach(() => {
    cleanup()
  })

  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(
        <Telephone label="My Telephone Label" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Telephone Label')
      expect(label.textContent).toBe('My Telephone Label')
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const { getByLabelText, container } = render(
        <Telephone label="My Label" initialValue="0800121212" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Label')
      expect(input.value).toBe('0800121212')
    })
  })
})
