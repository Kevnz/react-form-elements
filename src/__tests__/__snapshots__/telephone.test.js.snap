// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Telephone component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My Telephone Label
  </label>
  <input
    id="id-1-1"
    name="TextBox"
    type="tel"
    value=""
  />
</div>
`;

exports[`Telephone component handling input should set a value when passed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Label
  </label>
  <input
    id="id-2-1"
    name="TextBox"
    type="tel"
    value="0800121212"
  />
</div>
`;
